packages/app-desktop/gui/NoteEditor/utils/contextMenuUtils.ts

```diff
@@ -0,0 +1,85 @@
+import Resource from '@joplin/lib/models/Resource';
+
+export enum ContextMenuItemType {
+	None = '',
+	Image = 'image',
+	Resource = 'resource',
+	Text = 'text',
+	Link = 'link',
+}
+
+export interface ContextMenuOptions {
+	itemType: ContextMenuItemType;
+	resourceId: string;
+	mime: string;
+    filename: string;
+	linkToCopy: string;
+	textToCopy: string;
+	htmlToCopy: string;
+	insertContent: Function;
+	isReadOnly?: boolean;
+}
+
+export interface ContextMenuItem {
+	label: string;
+	onAction: Function;
+	isActive: Function;
+}
+
+export interface ContextMenuItems {
+	[key: string]: ContextMenuItem;
+}
+
+export async function resourceInfo(options: ContextMenuOptions): Promise<any> {
+	const resource = options.resourceId ? await Resource.load(options.resourceId) : null;
+	const filePath = resource ? Resource.fullPath(resource) : null;
+	const filename = resource ? (resource.filename ? resource.filename : resource.title) : options.filename ? options.filename : '';
+	const getCopyPath = () => {
+		if (options.textToCopy && options.mime) {
+			return textToDataUri(options.textToCopy, options.mime);
+		}
+		return filePath;
+	};
+	return { resource, filePath, filename, getCopyPath };
```

Comment:
I feel we're not making progress here. I've asked you to remove this getCopyPath from here. I can't follow your code anymore so I don't know how it shoudl be fixed.

Reply:
I’m not sure I understand, you mentioned in this [comment](https://github.com/laurent22/joplin/pull/6126#discussion_r804711851) to keep getCopyPath  as it is. 
> Actually don't change this one at all. It's used to get the resource info and nothing else.

Reply:
Ok there was some confusion. I meant to not change the resourceInfo function at all. Just leave it as it was because you are adding unrelated functionality to it. Your getCopyPath function shouldn't be here.